crates/ere-zisk/src/lib.rs

```diff
@@ -162,113 +100,14 @@ impl zkVM for EreZisk {
-        tempdir
-            .write_input(&input_bytes)
-            .map_err(|e| ZiskError::Prove(ProveError::TempDir(e)))?;
-
-        // Prove.
-
-        // TODO: Use `mpirun --np {num_processes} cargo-zisk prove ...` to
-        //       utilize multiple CPUs, probably need the `ProverResourceType`
-        //       to specify the number of available CPUs.
-
-        let start = time::Instant::now();
-        match self.resource {
-            ProverResourceType::Cpu => {
-                let status = Command::new("cargo-zisk")
-                    .arg("prove")
-                    .arg("--elf")
-                    .arg(tempdir.elf_path())
-                    .arg("--input")
-                    .arg(tempdir.input_path())
-                    .arg("--output-dir")
-                    .arg(tempdir.output_dir_path())
-                    .args(["--aggregation", "--verify-proofs", "--save-proofs"])
-                    .args(&self.prove_args)
-                    .status()
-                    .map_err(|e| ZiskError::Prove(ProveError::CargoZiskProve { source: e }))?;
-
-                if !status.success() {
-                    Err(ZiskError::Prove(ProveError::CargoZiskProveFailed {
-                        status,
-                    }))?
-                }
-            }
-            ProverResourceType::Gpu => {
-                // TODO: Set env `CUDA_VISIBLE_DEVICES = {0..num_devices}` to
-                //       control how many GPUs to use, probably need the `ProverResourceType`
-                //       to specify the number of available GPUs.
-                let witness_lib_path = dot_zisk_dir_path()
-                    .join("bin")
-                    .join("libzisk_witness_gpu.so");
-                let status = Command::new("cargo-zisk-gpu")
-                    .arg("prove")
-                    .arg("--witness-lib")
-                    .arg(witness_lib_path)
-                    .arg("--elf")
-                    .arg(tempdir.elf_path())
-                    .arg("--input")
-                    .arg(tempdir.input_path())
-                    .arg("--output-dir")
-                    .arg(tempdir.output_dir_path())
-                    .args(["--aggregation", "--verify-proofs", "--save-proofs"])
-                    .args(&self.prove_args)
-                    .status()
-                    .map_err(|e| ZiskError::Prove(ProveError::CargoZiskProve { source: e }))?;
-
-                if !status.success() {
-                    Err(ZiskError::Prove(ProveError::CargoZiskProveFailed {
-                        status,
-                    }))?
-                }
-            }
-            ProverResourceType::Network(_) => {
-                panic!(
-                    "Network proving not yet implemented for ZisK. Use CPU or GPU resource type."
-                );
-            }
-        }
-        let proving_time = start.elapsed();
+        let mut server = self.server()?;
```

Comment:
To double check understanding: so the idea is that in some future PR EreDockerized might leave a `ere-server-zisk` running, thus multiple `prove(...)` calls will hit this running instance where only the first `server.server()?` should actually spin up the whole thing, and the rest should only check status, no?

And I also suspect won't require volumes?

Reply:
Yes exactly, if the server don't crash we will only spin up the server once.

Volumes should then be optional because the global setup and rom setup are cached.